### PR TITLE
[Parse] Fix a crash when there is nothing after the prefix/postfix operator colon

### LINKED ISSUE #60932: "postfix operator ++:" -> Segmentation fault in compiler
The following line of code will crash the swift compiler:

postfix operator ++:

(the ":" is wrong and will produce the bug).

**Steps To Reproduce**
1. Open a new macOs console application in XCode
2. In the prepared source file, you can delete all lines and only have one line stating: "postfix operator ++:"

**Expected behavior**
I would expect the compiler to handle this incorrect code with an error, not with a segmentation fault

**Screenshots**
I can reproduce this issue on both an iMac and an MacBook Pro

**Environment (please fill out the following information)**
 - OS: macOS 12.3.1 (21E258)
 - Xcode Version/Tag/Branch: Version 13.4.1 (13F100)

**Additional context**
EmitSwiftModule normal x86_64 (in target 'CompilerCrash' from project 'CompilerCrash')
    cd /Users/user/Desktop/CompilerCrash
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /Users/user/Desktop/CompilerCrash/CompilerCrash/main.swift -target x86_64-apple-macos12.3 -enable-objc-interop -stack-check -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -I /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug -F /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug -enable-testing -g -module-cache-path /Users/user/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D DEBUG -new-driver-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -serialize-debugging-options -Xcc -working-directory -Xcc /Users/user/Desktop/CompilerCrash -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-generated-files.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-own-target-headers.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-all-target-headers.hmap -Xcc -iquote -Xcc /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-project-headers.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug/include -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources-normal/x86_64 -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources/x86_64 -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources -Xcc -DDEBUG\=1 -module-name CompilerCrash -target-sdk-version 12.3 -emit-module-doc-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftdoc -emit-module-source-info-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftsourceinfo -emit-objc-header-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash-Swift.h -o /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftmodule -emit-abi-descriptor-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.abi.json

/Users/user/Desktop/CompilerCrash/CompilerCrash/main.swift:5:1: error: expected precedence group name after ':' in operator declaration

^
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project and the crash backtrace.
Stack dump:
0.	Program arguments: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /Users/user/Desktop/CompilerCrash/CompilerCrash/main.swift -target x86_64-apple-macos12.3 -enable-objc-interop -stack-check -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -I /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug -F /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug -enable-testing -g -module-cache-path /Users/user/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity=checked -Onone -D DEBUG -new-driver-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -serialize-debugging-options -Xcc -working-directory -Xcc /Users/user/Desktop/CompilerCrash -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-generated-files.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-own-target-headers.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-all-target-headers.hmap -Xcc -iquote -Xcc /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/CompilerCrash-project-headers.hmap -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Products/Debug/include -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources-normal/x86_64 -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources/x86_64 -Xcc -I/Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/DerivedSources -Xcc -DDEBUG=1 -module-name CompilerCrash -target-sdk-version 12.3 -emit-module-doc-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftdoc -emit-module-source-info-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftsourceinfo -emit-objc-header-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash-Swift.h -o /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.swiftmodule -emit-abi-descriptor-path /Users/user/Library/Developer/Xcode/DerivedData/CompilerCrash-dptrbsbwmfvzcpefjsmarfubobsl/Build/Intermediates.noindex/CompilerCrash.build/Debug/CompilerCrash.build/Objects-normal/x86_64/CompilerCrash.abi.json
1.	Apple Swift version 5.6.1 (swiftlang-5.6.0.323.66 clang-1316.0.20.12)
2.	Compiling with the current language version
3.	While evaluating request ParseSourceFileRequest(source_file "/Users/user/Desktop/CompilerCrash/CompilerCrash/main.swift")
4.	With parser at source location: /Users/user/Desktop/CompilerCrash/CompilerCrash/main.swift:5:1
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000010e4b3ed7 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 39
1  swift-frontend           0x000000010e4b2f28 llvm::sys::RunSignalHandlers() + 248
2  swift-frontend           0x000000010e4b4530 SignalHandler(int) + 288
3  libsystem_platform.dylib 0x00007ff80107adfd _sigtramp + 29
4  libsystem_platform.dylib 0x00000000e5f1a270 _sigtramp + 18446603374421472400
5  swift-frontend           0x000000010a7839a3 swift::Parser::parseDecl(swift::OptionSet<swift::Parser::ParseDeclFlags, unsigned int>, bool, llvm::function_ref<void (swift::Decl*)>) + 38099
6  swift-frontend           0x000000010a7d9fc7 swift::Parser::parseBraceItems(llvm::SmallVectorImpl<swift::ASTNode>&, swift::BraceItemListKind, swift::BraceItemListKind, bool&) + 2087
7  swift-frontend           0x000000010a75591a swift::Parser::parseTopLevel(llvm::SmallVectorImpl<swift::Decl*>&) + 234
8  swift-frontend           0x000000010a7cff73 swift::SimpleRequest<swift::ParseSourceFileRequest, swift::SourceFileParsingResult (swift::SourceFile*), (swift::RequestFlags)12>::evaluateRequest(swift::ParseSourceFileRequest const&, swift::Evaluator&) + 803
9  swift-frontend           0x000000010a66d32e swift::ParseSourceFileRequest::OutputType swift::evaluateOrDefault<swift::ParseSourceFileRequest>(swift::Evaluator&, swift::ParseSourceFileRequest, swift::ParseSourceFileRequest::OutputType) + 1838
10 swift-frontend           0x000000010a6630cd swift::SourceFile::getTopLevelDecls() const + 125
11 swift-frontend           0x000000010a0e35fc swift::performImportResolution(swift::SourceFile&) + 1660
12 swift-frontend           0x000000010937d62d swift::CompilerInstance::performParseAndResolveImportsOnly() + 2237
13 swift-frontend           0x000000010937d6aa swift::CompilerInstance::performSema() + 26
14 swift-frontend           0x00000001091dbe8a swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 14234
15 swift-frontend           0x000000010919b5d4 swift::mainEntry(int, char const**) + 1108
16 dyld                     0x000000011318451e start + 462
error: Segmentation fault: 11 (in target 'CompilerCrash' from project 'CompilerCrash')

### PR DESCRIPTION
**Pull request by theblixguy**
Resolves https://github.com/apple/swift/issues/60932

### CODE REVIEW

**Comment by theblixguy** [comment]
@swift-ci please smoke test

File: validation-test/compiler_crashers_2_fixed/github_issue_60932.swift [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// RUN: not %target-typecheck-verify-swift
+
+// https://github.com/apple/swift/issues/60932
+
+// expected-error@+1 {{expected precedence group name after ':' in operator declaration}}
+prefix operator ++: // expected-error {{only infix operators may declare a precedence}} {{19-20=}}
```
Comment (AnthonyLatsis) [inline validation-test/compiler_crashers_2_fixed/github_issue_60932.swift:6]: This crasher is fairly simple, and we know the reason it crashes: should we add a proper test case instead?

**Review by AnthonyLatsis — COMMENTED** [review]
If you choose to keep the new test file, may I ask that you stick to the `issue-xxxxx` format? This is the one I am planning to use once I come around to rename files.

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Review by rintaro — APPROVED** [review]
Thank you!

**Pull request merged**